### PR TITLE
Pagination and image sizes for screenshots in UI Builds.

### PR DESCRIPTION
**Pull request by HelenaEksler**
#74
#92

### CODE REVIEW

**Comment by HelenaEksler** [comment]
shoov4

**Comment by HelenaEksler** [comment]
@amitaibu I've set 5 items per page. Tell me if I should change this. And could you please review the PR?

**Comment by amitaibu** [comment]
The items per page should be the one we get from restful

**Comment by HelenaEksler** [comment]
Range isn't set in restful. Maybe it would be better to add custom input to set Items per page? 
selection_381

**Comment by amitaibu** [comment]
Range is 50 by default (see $range). no need for range in the ui.

**Comment by amitaibu** [comment]
@solker can you pickup this one please.

File: shoov/modules/custom/shoov_screenshot/shoov_screenshot.module [hunk 1]
```diff
@@ -22,6 +22,13 @@ function shoov_screenshot_menu() {
     'type' => MENU_CALLBACK,
   );
 
+  $items['shoov/images/styles/%image_style'] = array(
```
Comment (HelenaEksler) [inline shoov/modules/custom/shoov_screenshot/shoov_screenshot.module:28]: @amitaibu This was missing. Using callback from image.module, that will create (of take existing) style images.

Comment (amitaibu) [inline shoov/modules/custom/shoov_screenshot/shoov_screenshot.module:28]: :+1:

Comment (amitaibu) [inline shoov/modules/custom/shoov_screenshot/shoov_screenshot.module:28]: maybe add a comment where it was copied form.

**Comment by HelenaEksler** [comment]
selection_384
selection_385
selection_386
selection_387

File: client/app/views/dashboard/screenshots/screenshots.html [hunk 2]
```diff
@@ -42,6 +42,18 @@
       </div>
     </div>
 
+    <div class="row">
+      <div class="col-md-4">
+        <label for="image-style">Image style:</label>
```
Comment (amitaibu) [inline client/app/views/dashboard/screenshots/screenshots.html]: => Image size

**Comment by HelenaEksler** [comment]
@amitaibu I think, we should create custom image styles with only width set (in order not to deform the images) and also change displaying images height based on chosen image style.

**Comment by amitaibu** [comment]
> we should create custom image styles with only width set

Good idea. And you can pass that list [here](https://github.com/shoov/shoov/blob/master/shoov/modules/custom/shoov_restful/plugins/restful/node/screenshots/1.0/ShoovScreenshotsResource.class.php#L24)

The height should indeed be by aspect ratio

**Comment by HelenaEksler** [comment]
selection_009
selection_011
selection_010
selection_012

**Comment by HelenaEksler** [comment]
@amitaibu could you review this, please?

**Comment by amitaibu** [comment]
@HelenaEksler tests failing

**Comment by HelenaEksler** [comment]
Yeah, I see. Trying to figure out why.

**Comment by HelenaEksler** [comment]
@amitaibu Tests are fixed. Could you review this PR?

File: client/app/scripts/controllers/screenshots.js [hunk 3]
```diff
@@ -8,15 +8,40 @@
...
 
-    // @todo: Change repo name to be taken from build.
-    $scope.repoName = screenshots[0].repository.label;
+    $scope.filteredScreenshots = [];
+    $scope.currentPage = 1;
+    $scope.itemsPerPage = 50;
+
+
+    $scope.imageStyles = {'self': 'Original'};
+    $scope.imageStyle = 'self';
```
Comment (amitaibu) [inline client/app/scripts/controllers/screenshots.js:26]: machine name should also be `original`

File: client/app/scripts/services/repos.js [hunk 4]
```diff
@@ -48,7 +48,7 @@ angular.module('clientApp')
      */
     function getDataFromBackend(repoId) {
       var deferred = $q.defer();
-      var url = Config.backend + '/api/github_repos';
+      var url = Config.backend + '/api/repositories';
```
Comment (amitaibu) [inline client/app/scripts/services/repos.js]: thanks, but wrong :)

File: client/app/scripts/controllers/screenshots.js [hunk 5]
```diff
@@ -8,15 +8,40 @@
...
+
+    if (screenshots.length) {
+      // Image styles for select list.
+      angular.forEach(screenshots[0].regression.styles, function(style, key) {
+        $scope.imageStyles[key] = style.label;
+      });
+    }
+
+
+    $scope.$watch('currentPage + numPerPage', function() {
```
Comment (amitaibu) [inline client/app/scripts/controllers/screenshots.js]: no need to watch `numPerPage`

File: client/app/scripts/controllers/screenshots.js (hunk 3 above)
Comment (amitaibu) [inline client/app/scripts/controllers/screenshots.js:26]: or, keep self and

`We use the "self" key..`

File: shoov/modules/custom/shoov_restful/plugins/restful/node/ShoovEntityBaseNode.php [hunk 6]
```diff
@@ -53,6 +53,16 @@ protected function imageProcess($value) {
       }
       return $output;
     }
+
+    $defaultStyles = shoov_screenshot_image_default_styles();
```
Comment (amitaibu) [inline shoov/modules/custom/shoov_restful/plugins/restful/node/ShoovEntityBaseNode.php]: this part looks wrong - why do we need it?

Comment (HelenaEksler) [inline shoov/modules/custom/shoov_restful/plugins/restful/node/ShoovEntityBaseNode.php]: We have this in order not to hardcode the styles. If later we'll decide to add/change/remove some image styles (e.g. add one more size or change the name of the existing one) - we won't have to change it in different places - just download the features and add them to the field in restful class.

Comment (amitaibu) [inline shoov/modules/custom/shoov_restful/plugins/restful/node/ShoovEntityBaseNode.php]: I'd prefer to harcode them in https://github.com/shoov/shoov/blob/fc643475ca0731685dcac3fe5a3a29424750b5e1/shoov/modules/custom/shoov_restful/plugins/restful/node/screenshots/1.0/ShoovScreenshotsResource.class.php#L24

Comment (amitaibu) [inline shoov/modules/custom/shoov_restful/plugins/restful/node/ShoovEntityBaseNode.php]: (and few lines below)

**Comment by amitaibu** [comment]
Merged, thanks.

**Pull request closed without merging**